table/block_based/block_based_table_reader.cc

```diff
@@ -4049,7 +4052,13 @@ Status BlockBasedTable::CreateIndexReader(
       std::unique_ptr<Block> metaindex_guard;
       std::unique_ptr<InternalIterator> metaindex_iter_guard;
       auto meta_index_iter = preloaded_meta_index_iter;
-      if (meta_index_iter == nullptr) {
+      bool should_fallback = false;
+      if (!rep_->internal_prefix_transform) {
```

Comment:
Do you think explicit nullptr check would be more readable than ! operator?

Reply:
Sure I can do that too.